include/no_os_tdm.h

```diff
@@ -107,8 +107,12 @@ struct no_os_tdm_platform_ops {
 				const struct no_os_tdm_init_param *);
 	/** TDM read operation function pointer */
 	int32_t (*tdm_ops_read)(struct no_os_tdm_desc *, void *, uint16_t);
+	/** TDM DMA read operation function pointer */
+	int32_t (*tdm_ops_read_dma)(struct no_os_tdm_desc *, void *, uint16_t);
```

Comment:
So these read_dma/write_dma ops are direct mappings of STM32 HAL in this no-OS top level include file. I think we shouldn't do this. In this file we need to be as generic as possible to cover all possible TDM peripherals and what we can do with them.

A better option would be to keep the read/write functions as they are but:
- add ops for stop()
- add ops for pause()
- add ops for resume()
- in no_os_tdm_init_param add callback for void tx_complete_callback(void *arg) and a void *tx_arg
- in no_os_tdm_init_param add callback for void rx_complete_callback(void *arg) and a void *rx_arg

All these new members can be left uninitialized and read()/write() will have normal blocking behavior.
However, if a callback is initialized, read()/write() should have non-blocking behavior.

Pause/Stop/Resume have no meaning in blocking mode so they can either silently return 0 or -EBADSTATE.

Reply:
Do you suggest something similar to the [UART implementation](https://github.com/analogdevicesinc/no-OS/blob/master/drivers/platform/stm32/stm32_uart.c#L130) here?

Reply:
Exactly !

Reply:
Added ops for stop/pause and resume explicitly.
Added a structure member for the callback functions as well.